pkg/daemon/rpm-ostree.go

```diff
@@ -28,7 +28,25 @@ type RpmOstreeDeployment struct {
 	CustomOrigin []string `json:"custom-origin"`
 }
 
-func getBootedDeployment(rootMount string) (*RpmOstreeDeployment, error) {
+// NodeUpdaterClient is an interface describing how to interact with the host
+// around content deployment
+type NodeUpdaterClient interface {
+	GetBootedDeployment(string) (*RpmOstreeDeployment, error)
+	GetBootedOSImageURL(string) (string, string, error)
+	RunPivot(string) error
+}
+
+// RpmOstreeClient provides all RpmOstree related methods in one structure.
+// This structure implements DeploymentClient
+type RpmOstreeClient struct{}
+
+// NewNodeUpdaterClient returns a new instance of the default DeploymentClient (RpmOstreeClient)
+func NewNodeUpdaterClient() *RpmOstreeClient {
+	return &RpmOstreeClient{}
+}
+
+// GetBootedDeployment returns the current deployment found
+func (r *RpmOstreeClient) GetBootedDeployment(rootMount string) (*RpmOstreeDeployment, error) {
```

Comment:
is this function called by anything external to this package? I only see the one change. if not, I would probably advocate for removing it from the client interface and just having it as a method on `RpmOstreeClient`.

Reply:
/cc @jlebon

Reply:
Updated.